File: backend/enum-jpa-querydsl-app/src/main/java/com/example/enumapp/domain/file/FileAssetRepository.java
package com.example.enumapp.domain.file;

import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;
import java.util.List;

public interface FileAssetRepository extends JpaRepository<FileAsset, Long> {

    List<FileAsset> findByStatusAndCreatedAtBefore(FileAssetStatus status, LocalDateTime threshold);
}

File: backend/enum-jpa-querydsl-kotlin-app/src/main/kotlin/com/example/enumapp/domain/file/FileAssetRepository.kt
package com.example.enumapp.domain.file

import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime

interface FileAssetRepository : JpaRepository<FileAsset, Long> {

    fun findByStatusAndCreatedAtBefore(status: FileAssetStatus, threshold: LocalDateTime): List<FileAsset>
}
